### DISCUSSION TITLE
An update about me and some suggestions

### DISCUSSION THREAD

**Discussion by samfundev** [twireapp/Twire#252] [General]
I haven't been active for the last few months and I wanted to let everyone know what happened. College started up for me and left me with way less time for me to work on Twire. Apologies for not saying something earlier, I should have added a new maintainer who could help out. Thanks to @Perflyst for adding another maintainer (@Nickwasused) who has been helping out. I'll still be trying to make contributions when I can.

Since we have multiple maintainers now, here are some suggestions. I'd be curious to hear from other maintainers about these suggestions.
 - Disable merge commits. I feel like they make a huge mess of the git history.
 - Consider looking for more maintainers.
 - Maintainers should submit PRs instead of committing.
 - PRs should be reviewed by someone before merging.
 
Also, the last two shouldn't be strict requirements. Small things that don't need review, like updating the gradle version, could be committed without review. I do think it is important though to have someone else just check over the code as there could be things that get missed. I will try to review PRs as well.

**Comment by Nickwasused** [comment]
> 
> 
> I haven't been active for the last few months and I wanted to let everyone know what happened. College started up for me and left me with way less time for me to work on Twire. Apologies for not saying something earlier, I should have added a new maintainer who could help out. Thanks to @Perflyst for adding another maintainer (@Nickwasused) who has been helping out. I'll still be trying to make contributions when I can.
> 

First of all, thank you for informing us about your current status.
I am coming right to the points you mentioned.

> Since we have multiple maintainers now, here are some suggestions. I'd be curious to hear from other maintainers about these suggestions.
> 
>     * Disable merge commits. I feel like they make a huge mess of the git history.
> 

I have looked into this in the Settings of another Repo I have and I think you mean the setting here: 

![firefox_7kuYRJcwlO](https://user-images.githubusercontent.com/34883496/136692606-1b6fc602-0785-4aa0-b5e9-3bae7733640f.png)
I have nothing against it. It sounds good to have a clean git history.

>     * Consider looking for more maintainers.
> 

More Maintainers are always welcome and needed.

>     * Maintainers should submit PRs instead of committing.
> 

Sounds good to me.

>     * PRs should be reviewed by someone before merging.
> 

Someone should always look over the code, especially if new Stuff gets added or if there are major changes.

> 
> Also, the last two shouldn't be strict requirements. Small things that don't need review, like updating the gradle version, could be committed without review. I do think it is important though to have someone else just check over the code as there could be things that get missed. I will try to review PRs as well.

~~As you mentioned the last two being strict requirements there is an option for that. I think you can set the Number of users that need to approve the Changes. (https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/managing-a-branch-protection-rule)~~
(Let's just pretend like there is no text above xD)

**Reply by Perflyst** [reply]
> As you mentioned the last two being strict requirements there is an option for that.

They mentioned "Also, the last two **shouldn't** be strict requirements."  ;)
